pkg/util/uri_sanitize.go

```diff
@@ -0,0 +1,72 @@
+package util
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+const masking = "hidden"
+
+var sensitiveQueryChecks = map[string]func(v map[string]string) bool{
+	"auth_token": func(v map[string]string) bool {
+		if _, ok := v["auth_token"]; ok {
+			return true
+		}
+		return false
+	},
+	"x-amz-signature": func(v map[string]string) bool {
+		if _, ok := v["x-amz-signature"]; ok {
+			return true
+		}
+		return false
+	},
+	"x-goog-signature": func(v map[string]string) bool {
+		if _, ok := v["x-goog-signature"]; ok {
+			return true
+		}
+		return false
+	},
+	"sig": func(v map[string]string) bool {
+		if _, ok := v["sig"]; !ok {
+			return false
+		}
+		if _, ok := v["sv"]; ok {
+			return true
+		}
+		return false
+	},
+}
+
+func lowerToKeyMap(values url.Values) map[string]string {
+	lm := make(map[string]string)
+	for key := range values {
+		lm[strings.ToLower((key))] = key
+	}
+	return lm
+}
+
+func SanitizeURI(s string) (string, error) {
+	if s == "" {
+		return s, nil
+	}
+
+	u, err := url.ParseRequestURI(s)
+	if err != nil {
+		return "", fmt.Errorf("failed to sanitize URL")
+	}
+
+	// strip out sensitive query strings
+	urlValues := u.Query()
+	keys := lowerToKeyMap(urlValues)
+	for key, checker := range sensitiveQueryChecks {
+		if originalKey, ok := keys[key]; ok {
+			if checker(keys) {
+				urlValues.Set(originalKey, masking)
+			}
+		}
+	}
+	u.RawQuery = urlValues.Encode()
+
+	return u.String(), nil
+}
```

Comment:
This seems like it may be overly complex, and also suffers from a problem if there are multiple variants of the same key that differ only by case, where it will arbitrarily mask one and ignore the other depending on the order of the iteration in lowerToKeyMap (because the last-seen original key will be the one in `lm` for a given lowercase key).

It may be simpler overall to say something like this:
```suggestion
func SanitizeURI(s string) (string, error) {
	if s == "" {
		return s, nil
	}

	u, err := url.ParseRequestURI(s)
	if err != nil {
		return "", fmt.Errorf("failed to sanitize URL")
	}

	// strip out sensitive query strings
	urlValues := u.Query()
	for key, values := range urlValues {
		if checker, ok := sensitiveQueryChecks[strings.ToLower(key)]; ok {
			if checker(values, urlValues) {
				urlValues.Set(key, masking)
			}
		}
	}
	u.RawQuery = urlValues.Encode()

	return u.String(), nil
}
```
If the signature of checker is updated to `func(values []string, urlValues url.Values) bool` then the simple cases can be very simple indeed; `return true`, and the complex case can handle its own case insensitivity (by looping through urlValues instead of trying to look up by lowercase key).

Reply:
Thanks. If it gets to complex with case-insensitive lets punt on this for the first version and revisit if needed later

Reply:
@DanCech thanks for reviewing this one! I will look at your suggestion and update the PR 🙏🏻